src/helpers/iconpicker.class.js

```diff
@@ -0,0 +1,173 @@
+    block.trigger('picture:change', pictureInformations);
+
+    block.modal.close();
+
+}
+
+function bindClickOnIcons(block) {
+    $.each(block.slider.$elem.find('img'), function(){
+        $(this).on('click', function(){
+            triggerChangeIllustratedPicture(block, {
+                src: $(this).attr('src'),
+                copyright: $(this).attr('alt'),
+                id: $(this).data('id')
+            });
+        });
+    });
+}
+
+function dropEvent(iconPicker){
+    $(iconPicker.modal.$elem.children('.droppable')[0]).off();
+    $(iconPicker.modal.$elem.children('.droppable')[0]).on('dragover dragenter', function(ev){
+        ev.preventDefault();
+        ev.stopPropagation();
+    });
+
+
+    $(iconPicker.modal.$elem.children('.droppable')[0]).on('drop', function(ev){
+        ev.preventDefault();
+        ev.stopPropagation();
+
+        iconPicker.blockRef.loading();
+
+        var file = ev.originalEvent.dataTransfer.files[0];
+
+        if (/image/.test(file.type)) {
+
+            var fileUploader = new FileUploader(iconPicker.blockRef, iconPicker.blockRef.globalConfig.apiUrl + 'edt/media/upload');
+
+            var uploadedMedia;
+
+            var uploadPromise = fileUploader.upload(file)
+            .then(function(returnedData){
+                uploadedMedia = iconPicker.blockRef.globalConfig.apiUrl + 'edt/media/' + returnedData.idMedia;
+            });
+
+            uploadPromise.then(function(){
+                iconPicker.blockRef.ready();
+                iconPicker.modal.close();
+                xhr.get(uploadedMedia)
+
+                .then(function(imageData) {
+                   triggerChangeIllustratedPicture(iconPicker, {
+                        src: imageData.content.thumbnail,
+                        copyright: imageData.content.copyright,
+                        id: imageData.content.id
+                    });
+                });
+            });
+
+            uploadPromise.catch(function(err){
+                console.log(err);
+            });
+        }
+    });
+}
+
+function createArrayOfIcons(icons) {
+    var iconsArray = [];
+    Object.keys(icons).forEach(function(k) {
+        var file = icons[k].file.replace('original', '90x90');
+        var single = _.template('<img data-id=<%= id %> src="<%= icon %>" alt="<%= alt %>" >', { icon: file, alt: icons[k].legend, id: icons[k].id });
+        iconsArray.push(single);
+    });
+    return iconsArray;
+}
+
+function getIcons(url, block) {
+    xhr.get(url)
+    .then(function(iconData) {
+        var iconsArray = createArrayOfIcons(iconData.content);
+
+        var modalInner = block.modal.$elem.children('.modal-content')[0];
+
+        if (block.modal.$elem.children('.icon-picker') === false) {
+            block.iconPicker = iconPickerHtml;
+            $(modalInner).append(block.iconPicker);
+            block.modal.$elem.children('.icon-picker').addClass('is-visible');
+        }
+        dropEvent(block);
```

Comment:
@arthur-balagne functions should not do 10,000 things. there is no need for dropEvent to be here